tests/test_cli_compile.py

```diff
@@ -839,3 +839,27 @@ def test_empty_input_file_no_header(runner, empty_input_pkg, prior_output_pkg):
 
     with open("requirements.txt", "r") as req_txt:
         assert req_txt.read().strip() == ""
+
+
+def test_upgrade_package_doesnt_remove_annotation(pip_conf, runner):
+    """
+    Tests pip-compile --upgrade-package shouldn't remove "via" annotation.
+    See: GH-929
+    """
+    with open("requirements.in", "w") as req_in:
+        req_in.write("small-fake-with-deps-a\n")
+
+    runner.invoke(cli)
+
+    with open("requirements.txt", "r") as req_txt:
+        assert (
+            "small-fake-a==0.1         # via small-fake-with-deps-a"
+            in req_txt.read().splitlines()
+        )
+
+    runner.invoke(cli, ["-P", "small-fake-a"])
+    with open("requirements.txt", "r") as req_txt:
+        assert (
+            "small-fake-a==0.1         # via small-fake-with-deps-a"
```

Comment:
Can we test here that `small-fake-a` has been also upgraded to a different version?

I guest that might take change `small-fake-with-deps-a` dependencies, so we can fix it by:
1. pin small-fake-a==0.2 in https://github.com/jazzband/pip-tools/blob/8e782ae5051ef8e195bb77e1955fed3f2c607fd5/tests/test_data/packages/small_fake_with_deps_a/setup.py#L4
2. fix `test_annotate_option` to honor that change
3. update the last assertion in `test_upgrade_package_doesnt_remove_annotation`

Reply:
Probably better to do it in a separate commit, to preserve the intention to update `small-fake-with-deps-a` in `git blame`.

Reply:
Good idea. when I do this change `small-fake-a==0.2`, the `test_annotate_option` won't fail.
I don't know why? is it something related to my machine?

Reply:
Interesting... Push the commit -- we'll figure out.

Reply:
Oh, I see, you should rebuild `tests/test_data/minimal_wheels/small_fake_with_deps_a-0.1-py2.py3-none-any.whl`.

Reply:
I did the rebuild. and upgrade `small-fake-with-deps-a` dependencies to `small-fake-a==0.2`.
Also, changed `test_annotate_option`. I did them in a separate commit.

Seems `runner.invoke(cli, ["-P", "small-fake-a"])` didn't work properly and the `small-fake-a` didn't upgraded.

Am I wrong?

Reply:
I'll resolve this, since it looks like you've figured out how to fix this.